src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -145,3 +159,16 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
         return !!version && gte(version, '11.0.0');
     }
 }
+
+async function validateDatabaseName(name: string, getChildrenTask: Promise<AzExtTreeItem[]>): Promise<string | undefined | null> {
+    if (!name || name.length < 1) {
+        return localize('NameCannotBeEmpty', 'Name cannot be empty.');
+    }
+    const currDatabaseList = await getChildrenTask;
+    const currDatabaseNames: string[] = [];
+    currDatabaseList.map(database => currDatabaseNames.push(database.label));
```

Comment:
As a general rule, `label` should only be used for display purposes. We've had bugs in the past because `label` isn't always the same as name. In this case, I would suggest something like this:
```
for (const db of currDatabaseList) {
    if (db instanceof PostgresDatabaseTreeItem) {
        currDatabaseNames.push(db.databaseName);
    }
}
```
I'm using `databaseName` instead of `label` and the `instanceof` check helps in the rare case where the children includes an unexpected tree item. For example, `createTreeItemsWithErrorHandling` can include an `InvalidTreeItem` if one child has an error, but the rest are fine